color_detect/color_detect.py

```diff
@@ -18,128 +18,128 @@
+	return (knownWidth * focalLength) / perWidth
 
 # 计算摄像头的焦距（内参）
 def calculate_focalDistance(video):    
-    success,image = video.read()
-    marker = find_marker(image)       
-    print("图片中A4纸的宽度：f%", marker[1][0])
-    focalLength = (marker[1][0] * KNOWN_DISTANCE) / KNOWN_WIDTH  
-    print('焦距 = ', focalLength)        
-    return focalLength
+	success,image = video.read()
+	marker = find_marker(image)       
+	print("图片中A4纸的宽度：f%", marker[1][0])
+	focalLength = (marker[1][0] * KNOWN_DISTANCE) / KNOWN_WIDTH  
+	print('焦距 = ', focalLength)        
+	return focalLength
 
 
 # 计算摄像头到物体的距离
 def calculate_Distance(focalLength_value):
-    success,image = video.read()
-    # 获取矩形的中心点坐标，长度，宽度和旋转角度， marke[1][0]代表宽度
-    marker = find_marker(image)     
-    distance_cm = distance_to_camera(KNOWN_WIDTH, focalLength_value, marker[1][0])
-    box = cv2.boxPoints(marker)
-    # print("Box = ", box)
-    
-    center_x = (box[0][0]+box[1][0]+box[2][0]+box[3][0])/4
-    center_y = (box[0][1]+box[1][1]+box[2][1]+box[3][1])/4
-    print("x_diff:",center_x-320)
-    x_diff = center_x-320
-    
-    box = np.int0(box)
-    # print("Box = ", box)
-    cv2.circle(image, (int(center_x),int(center_y)), 3, (1, 227, 254), -1)
-    cv2.drawContours(image, [box], -1, (0, 255, 0), 2)
-    cv2.putText(image, "%.2fcm" % (distance_cm),
-            (image.shape[1] - 300, image.shape[0] - 20), cv2.FONT_HERSHEY_SIMPLEX,
-            2.0, (0, 0, 255), 3)          
-    cv2.imshow("image", image)
-    # waitKey(30)
-    print(distance_cm,"cm")
-    return distance_cm,x_diff
+	success,image = video.read()
+	# 获取矩形的中心点坐标，长度，宽度和旋转角度， marke[1][0]代表宽度
+	marker = find_marker(image)     
+	distance_cm = distance_to_camera(KNOWN_WIDTH, focalLength_value, marker[1][0])
+	box = cv2.boxPoints(marker)
+	# print("Box = ", box)
+	
+	center_x = (box[0][0]+box[1][0]+box[2][0]+box[3][0])/4
+	center_y = (box[0][1]+box[1][1]+box[2][1]+box[3][1])/4
+	print("x_diff:",center_x-320)
+	x_diff = center_x-320
+	
+	box = np.int0(box)
+	# print("Box = ", box)
+	cv2.circle(image, (int(center_x),int(center_y)), 3, (1, 227, 254), -1)
+	cv2.drawContours(image, [box], -1, (0, 255, 0), 2)
+	cv2.putText(image, "%.2fcm" % (distance_cm),
+			(image.shape[1] - 300, image.shape[0] - 20), cv2.FONT_HERSHEY_SIMPLEX,
+			2.0, (0, 0, 255), 3)          
+	cv2.imshow("image", image)
+	# waitKey(30)
+	print(distance_cm,"cm")
+	return distance_cm,x_diff
 
 
 def color_detect_server():
-    rospy.init_node = (color_detect_server)
-    s = rospy.service("color_detect",Int16MutiArray,main0)
-    rospy.spin()
+	rospy.init_node = (color_detect_server)
+	s = rospy.service("color_detect",Int16MutiArray,main0)
+	rospy.spin()
 
 def main0(req):
-    success,image = video.read()
-    color = find_blue_object(image)
-    color = find_orange_object(image)
-    color = find_black_object(image)
+	success,image = video.read()
+	color = find_blue_object(image)
+	color = find_orange_object(image)
+	color = find_black_object(image)
```

Comment:
`color` variable from line `131` to `132` is replaced by line `133`. You might want to change the way of implementing this.